graphbolt/src/csc_sampling_graph.cc

```diff
@@ -95,6 +95,48 @@ void CSCSamplingGraph::Save(torch::serialize::OutputArchive& archive) const {
   }
 }
 
+void CSCSamplingGraph::SetState(
+    torch::Dict<std::string, torch::Dict<std::string, torch::Tensor>>& state) {
+  auto&& regular_tensors = state.at("regular_tensors");
+  TORCH_CHECK(
+      regular_tensors.at("magic_number").equal(torch::tensor({1})),
+      "Magic number mismatches when loading pickled CSCSamplingGraph.")
+  indptr_ = regular_tensors.at("indptr");
+  indices_ = regular_tensors.at("indices");
+  if (regular_tensors.find("node_type_offset") != regular_tensors.end()) {
+    node_type_offset_ = regular_tensors.at("node_type_offset");
+  }
+  if (regular_tensors.find("type_per_edge") != regular_tensors.end()) {
+    type_per_edge_ = regular_tensors.at("type_per_edge");
+  }
+  if (state.find("edge_attributes") != state.end()) {
+    edge_attributes_ = state.at("edge_attributes");
+  }
+}
+
+torch::Dict<std::string, torch::Dict<std::string, torch::Tensor>>
+CSCSamplingGraph::GetState() const {
+  // State is a vector of dicts. The first dict contains all the tensor
+  // attributes. If edge_attributes is not None, it will be stored as the second
+  // dict.
```

Comment:
what will happen in the future if we add another dict of tensor in CSCSamplingGraph?

Reply:
It seems that the comment if out of data? The state is a dict of dict in your code.

Reply:
The comment is updated now.
To add a new dict of tensor, put it into the state dict directly with its name as key. To add a new tensor, put it into the sub-dict with key `"regular_tensors"`.